src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/BaseFormattingRule.cs

```diff
@@ -32,26 +32,14 @@ protected static void AddUnindentBlockOperation(
     protected static void AddUnindentBlockOperation(
         List<IndentBlockOperation> list,
         SyntaxToken startToken,
-        SyntaxToken endToken,
-        bool includeTriviaAtEnd = false,
-        IndentBlockOption option = IndentBlockOption.RelativePosition)
+        SyntaxToken endToken)
     {
         if (startToken.Kind() == SyntaxKind.None || endToken.Kind() == SyntaxKind.None)
-        {
             return;
-        }
 
-        if (includeTriviaAtEnd)
-        {
-            list.Add(FormattingOperations.CreateIndentBlockOperation(startToken, endToken, indentationDelta: -1, option: option));
-        }
-        else
-        {
-            var startPosition = CommonFormattingHelpers.GetStartPositionOfSpan(startToken);
```

Comment:
the issue was this line.  the indent of the label is not based on the indent of the preceding token (which is what this was doing).  It's based on surrrounding context (which is determined from teh label start token itself, not the preceding token, which might be in a deeper nesting context).